internal/unpackinfo/filemode.go

```diff
@@ -0,0 +1,94 @@
+package unpackinfo
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+)
+
+type FileMode uint32
+
+const (
+	Empty FileMode = 0
+	// Dir represent a Directory.
+	Dir FileMode = 0040000
+	// Regular represent non-executable files.  Please note this is not
+	// the same as golang regular files, which include executable files.
```

Comment:
Hmm. Do you think it's worth using a slightly different name than Go does to avoid confusion? If so, I suggest "Plain".